htsohm/material_files.py

```diff
@@ -197,131 +179,95 @@ def write_child_definition_files(run_id, parent_id, generation, mutation_strengt
-        write_mixing_rules(mix_file, atom_types)
-        print('  - force_field_mixing_rules.def\tWRITTEN.')
+    # perturb LJ-parameters
+    child_material.atom_types = []
+    for atom_type in parent_material.atom_types:    
+        new_atom_type = {'chemical-id' : atom_type['chemical-id']}
+        new_atom_type['epsilon'] = (round(atom_type['epsilon'] +
+                mutation_strength * (uniform(*epsilon_limits) -
+                atom_type['epsilon']), 4))
+        new_atom_type['sigma'] = (round(atom_type['sigma'] +
+                mutation_strength * (uniform(*epsilon_limits) -
+                atom_type['epsilon']), 4))
+        child_material.atom_types.append(new_atom_type)
 
     ########################################################################
-    # load values from parent cif-file
-    p_cif = os.path.join(mat_dir, run_id + '-' + str(parent.uuid) + '.cif')
-    n1, n2, old_x, old_y, old_z = np.genfromtxt(p_cif, unpack=True, skip_header=16)
-    old_atom_types = np.genfromtxt(p_cif, usecols=0, dtype=str, skip_header=16)
-    old_abc = np.genfromtxt(
-        p_cif, unpack=True, skip_header=4, skip_footer=len(old_x) + 9, usecols = 1
-    )
-    ########################################################################
     # calculate new lattice constants
-    old_a = old_abc[0]
-    old_b = old_abc[1]
-    old_c = old_abc[2]
-    random_a = round(uniform(*lattice_limits), 4)
-    random_b = round(uniform(*lattice_limits), 4)
-    random_c = round(uniform(*lattice_limits), 4)
-    a = round(old_a + mutation_strength * (random_a - old_a), 4)
-    b = round(old_b + mutation_strength * (random_b - old_b), 4)
-    c = round(old_c + mutation_strength * (random_c - old_c), 4)
-    lattice_constants = {"a" : a, "b" : b, "c" : c}
+    child_material.lattice_constants = {}
+    for i in ['a', 'b', 'c']:
+        old_x = parent_material.lattice_constants[i]
+        random_x = uniform(*lattice_limits)
+        new_x = round(old_x + mutation_strength * (random_x - old_x), 4)
+        child_material.lattice_constants[i] = new_x
+
     ########################################################################
     # calulate new number density, number of atoms
-    old_ND = len(old_x) / (old_a * old_b * old_c)
-    random_ND = round(uniform(*number_density_limits), 4)
-    number_density = old_ND + mutation_strength * (random_ND - old_ND)
-    number_of_atoms = int(number_density * a * b * c)
+    old_LCs = parent_material.lattice_constants
+    old_vol = old_LCs['a'] * old_LCs['b'] * old_LCs['c']
+    old_number_density = len(parent_material.atom_sites) / old_vol
+    random_number_density = uniform(*number_density_limits)
+    new_number_density = (old_number_density + mutation_strength * (
+            random_number_density - old_number_density))
+    new_LCs = child_material.lattice_constants
+    new_vol = new_LCs['a'] * new_LCs['b'] * new_LCs['c']
+    number_of_atom_sites = int(new_number_density * new_vol)
+
```

Comment:
Since PsuedoMaterial is a class now, maybe we should load the yaml for the parent straight into an instance of PsuedoMaterial. Then you could add methods like density() and volume() and number_of_atoms() which would get calculated straight off the instance variables. Then I think this code could simplify to something like: 

random_number_density = uniform(*number_density_limits)
    new_number_density = parent.old_number_density + mutation_strength * (random_number_density - parent.old_number_density)
    child_material.number_of_atoms = int(new_number_density * child_material.volume())

Probably we could simplify a lot of this method by consolidating the logic into your PseudoMaterial class....